xdsl/dialects/riscv.py

```diff
@@ -373,7 +406,7 @@ def __init__(
         comment: str | StringAttr | None = None,
     ):
         if isinstance(immediate, int):
-            immediate = IntegerAttr.from_int_and_width(immediate, 32)
+            immediate = IntegerAttr(immediate, IntegerType(20, Signedness.SIGNED))
```

Comment:
I'm wondering whether we might want to convert all parameters to this bitwidth. So if we get another integer attr, we'd still catch whether the bitwidth is correct. This should possibly be done at the rewrite level, though.